lib/dor/models/editable.rb

```diff
@@ -232,16 +232,51 @@ def use_license=(use_license_machine)
     # @return [String] A description of the rights defined in the default object rights datastream. Can be 'Stanford', 'World', 'Dark' or 'None'
     def default_rights
       xml = defaultObjectRights.ng_xml
-      if xml.search('//rightsMetadata/access[@type=\'read\']/machine/group').length == 1
-        'Stanford'
-      elsif xml.search('//rightsMetadata/access[@type=\'read\']/machine/world').length == 1
-        'World'
-      elsif xml.search('//rightsMetadata/access[@type=\'discover\']/machine/none').length == 1
-        'Dark'
+      machine_read_access = xml.search('//rightsMetadata/access[@type="read"]/machine')
+      machine_discover_access = xml.search('//rightsMetadata/access[@type="discover"]/machine')
+
+      machine_read_access_node = machine_read_access.length == 1 ? machine_read_access.first : nil
+      machine_discover_access_node = machine_discover_access.length == 1 ? machine_discover_access.first : nil
+
+      if machine_read_access_node && machine_read_access_node.search('./group[text()="Stanford" or text()="stanford"]').length == 1
+        if machine_read_access_node.search('./group[@rule="no-download"]').length == 1
+          'stanford-nd'
+        else
+          'stanford'
+        end
+      elsif machine_read_access_node && machine_read_access_node.search('./world').length == 1
+        if machine_read_access_node.search('./world[@rule="no-download"]').length == 1
+          'world-nd'
+        else
+          'world'
+        end
+      elsif machine_read_access_node && machine_read_access_node.search('./location[text()="spec"]').length == 1
+        'loc:spec'
+      elsif machine_read_access_node && machine_read_access_node.search('./location[text()="music"]').length == 1
+        'loc:music'
+      elsif machine_discover_access_node && machine_discover_access_node.search('./world').length == 1
+        'none'
+      elsif machine_discover_access_node && machine_discover_access_node.search('./none').length == 1
+        'dark'
```

Comment:
ah, sorry, specifically in 257-260:  if none of the prior three top level branches are true (it's not a stanford restricted object, it's not a world readable object, it's not a location restricted object), then it must be either "citation only" or "dark".  if it's world discoverable, it's citation only.  if it's not discoverable, it's dark.  if the XML doesn't fit any of those patterns, it's in a structure we don't support/understand, and we can't return a corresponding rights type code.